python/src/aac/aac.aac

```diff
@@ -0,0 +1,1265 @@
+  name: typeref
+  package: aac.lang
+  python_type: str
+  description:  |
+    A reference to another defined type.  Referenced type are typed and documented 
+    as 'typeref(package_name.type_name)' denoting the package and type being referenced.
+    A typeref respects inheritance and will respect references using polymorphism.  For example,
+    you can reference primitive type, enums, and schemas by using the typeref 'typeref(aac.lang.AacType)'.
+    Because Primitive, enum, and Schema all extend from AacType, any value provided for any of these
+    types will be accepted by the typeref.
+  constraints:
+    - name: Check typeref
+---
+schema:
+  name: Schema
+  extends: 
+    - package: aac.lang
+      name: AacType
+  package: aac.lang
+  root: schema
+  description: |
+    A definition that defines the schema/structure of data.
+  fields:
+    - name: extends
+      type: SchemaExtension[]
+      description: |
+        A list of Schema definition names that this definition inherits from.
+    - name: modifiers
+      type: dataref(modifier.name)[]
+      description:
+        A means of further defining the schema and how it can be used within the model.
+    - name: root
+      type: string
+      description: |
+        The root key to use when declaring an instance of the type in yaml/aac files.
+    - name: fields
+      type: Field[]
+      description: |
+        A list of fields that make up the definition.
+      is_required: true
+    - name: requirements
+      type: dataref(req.id)[]
+      description: |
+        A list of requirements associated with this schema.
+    - name: constraints
+      type: SchemaConstraintAssignment[]
+---
+schema:
+  name: SchemaExtension
+  package: aac.lang
+  description: |
+    References the inheritance parent schema.
+  fields:
+    - name: package
+      type: dataref(schema.package)
+      description: |
+        The package name of the inheritance parent schema.
+      is_required: true
+    - name: name
+      type: dataref(schema.name)
+      description: |
+        The name of the inheritance parent schema.
+      is_required: true
+---
+schema:
+  name: Field
+  package: aac.lang
+  fields:
+    - name: name
+      type: string
+      description: |
+        The name of the field.
+
+        Used to refer to the field, in certain cases.
+      is_required: true
+    - name: type
+      type: typeref(aac.lang.AacType)
+      description: |
+        The type of the field.
+
+        Used to constrain the field's values to make it simpler to reason about. Must
+        reference an existing type name as declared in a schema.
+      is_required: true
+    - name: description
+      type: string
+      description: |
+        A description for the field to let users know what it's for and any
+        other useful information.
+    - name: is_required
+      type: bool
+      description: |
+        Indicates that the field must be present when used.
+      default: "false"
```

Comment:
This new field is supposed to have replaced the `required_fields` validator, so the logic for parsing, interpreting, and properly handling the field need to be evaluated against the task: #734

Reply:
Once the `LanguageError` occurrences that are missing the `location` arguments have been corrected (I just added a `None` for testing), this will be functioning as desired, see below, for providing the evaluation of required fields being present that the legacy validator provided. Which having the location data should allow for the returned notice to be more helpful. 

```
(venv) gitpod /workspace/AaC/python/features/alarm_clock (pure-py-api) $ aac check alarm_clock.yaml 
Missing required field model.
``` 

I believe the only portion missing is explicit documentation around this being a schema type constraint. It is called out as part of the `Checking an AaC File` in being referenced as a "core constraint that are automatically checked while parsing and loading the AaC definition". But as previously commented, the team had previously decided that as part of the validation to constrainer refactor work, we had wanted to incorporate explicit documentation on each one.